com.hybris.hyeclipse.commons/src/com/hybris/hyeclipse/commons/utils/EclipseFileUtils.java

```diff
@@ -0,0 +1,138 @@
+ */
+public final class EclipseFileUtils {
+
+	/**
+	 * Private in order to avoid class initialization
+	 */
+	private EclipseFileUtils() {
+		/* Intentionally empty */ }
+
+	/**
+	 * Returns selected file
+	 * 
+	 * @param selection
+	 *            current selection
+	 * @return selected {@link IFile}
+	 */
+	public static IFile getSelectedFile(final ISelection selection) {
+		IFile file = null;
+		
+		if (selection instanceof IStructuredSelection) {
+			final IStructuredSelection ssel = (IStructuredSelection) selection;
+			final Object obj = ssel.getFirstElement();
+			file = (IFile) Platform.getAdapterManager().getAdapter(obj, IFile.class);
+		} else if (selection instanceof TextSelection) {
+			final IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+			file = (IFile) window.getActivePage().getActiveEditor().getEditorInput()
+			                .getAdapter(IFile.class);
+		}
+		return file;
+	}
+
+	/**
+	 * Return set of selected file(s)
+	 * 
+	 * @param selection
+	 *            current selection
+	 * @return set of selected files
+	 */
+	@SuppressWarnings("unchecked")
+	public static Set<IFile> getSelectedFiles(final ISelection selection) {
+		final Set<IFile> files = new HashSet<>();
+
+		if (selection instanceof IStructuredSelection) {
+			final IStructuredSelection structuredSelection = (IStructuredSelection) selection;
+			structuredSelection.toList().forEach(
+			                file -> files.add((IFile) Platform.getAdapterManager().getAdapter(file, IFile.class)));
+		} else if (selection instanceof TextSelection) {
+			final IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+			files.add((IFile) window.getActivePage().getActiveEditor().getEditorInput().getAdapter(IFile.class));
+
+		}
+
+		return files;
+	}
+
+	/**
+	 * Returns optional of current text selection
+	 *  
+	 * @return optional of current text selection
+	 */
+	public static Optional<TextSelection> getCurrentTextSelection() {
+		Optional<TextSelection> currentTextSelection = Optional.empty();
+		final IEditorPart editorPart = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage()
+		                .getActiveEditor();
+		
+		if ( editorPart != null && editorPart instanceof ITextEditor) {
+			final ITextEditor editor = (ITextEditor) editorPart;
+			final ISelection selection = editor.getSelectionProvider().getSelection();
+			if (selection instanceof TextSelection) {
+				currentTextSelection = Optional.of((TextSelection) selection);
+			}
+		}
+		
+		return currentTextSelection;
+	}
+
+	/**
+	 * Returns selected text in file
+	 * 
+	 * @return selected text in file, if none is selected empty string will be returned
+	 */
+	public static String getCurrentSelectedText() {
+		final Optional<TextSelection> currentTextSelection = getCurrentTextSelection();
+		return (currentTextSelection.isPresent()) ? currentTextSelection.get().getText() : Constants.EMPTY_STRING;
```

Comment:
Change to currentSelection.map(TextSelection::getText).orElse(Constants.EMPTY_STRING) avoid using *get* method.